.eslintrc

```diff
@@ -0,0 +1,30 @@
+{
+    "parser": "@typescript-eslint/parser",
+    "extends": [
+        "airbnb",
```

Comment:
Looks like there's a lot going on here. I have my linter set up for only airbnb and it's catching a lot of formatting issues with your code. Almost entirely spaces. If something else comes up I'll mention it specifically.